### ISSUE TITLE
Parsing failure when using column named DELTA (reserved keyword) in Snowflake dialect

### ISSUE THREAD

**Issue by bressanmarcos** [sqlfluff/sqlfluff#7030]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I encountered a parsing error in SQLFluff when creating a table with a column named DELTA, without quoting. I'm using the Snowflake dialect.

```sql
-- .sqlfluff config:
# [sqlfluff]
# dialect = snowflake

CREATE TABLE IF NOT EXISTS my_schema.my_table (
  DELTA INT
);
```

Running:
```
sqlfluff lint my_table.sql --dialect snowflake
```

Produces:
```
L: 1 | P: 10 | PRS | Found unparsable section: '(\n  DELTA NUMB...'
```

Workaround:
---

Quoting the column name:
```
"DELTA" NUMBER
```

Context & Link to similar behavior:
---
This is the same class of issue documented in #5420, which describes failures when using reserved words like LIMIT or END as column names. DELTA appears to be similarly treated as a reserved word in the parser, despite being allowed when quoted.

### Expected Behaviour

Either:

- DELTA should be accepted as a valid identifier (with or without warning), or
- The parser should emit a clear error message indicating a reserved-word-as-identifier issue.

### Observed Behaviour

A vague parsing failure (Found unparsable section) with no hint that DELTA is causing the problem.

### How to reproduce

- Create a SQL file with the CREATE TABLE above.

- Lint with sqlfluff lint.

- Observe parse error.

- Change DELTA to "DELTA" (quoted), and the error disappears.

### Dialect

snowflake

### Version

3.4.1

### Configuration

[sqlfluff]
dialect = snowflake

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
